src/d3a/d3a_core/simulation.py

```diff
@@ -375,21 +375,20 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                 self.simulation_config.external_redis_communicator.\
                     publish_aggregator_commands_responses_events()
 
-                realtime_tick_length = time.time() - simulation_time_counter
                 if d3a.constants.RUN_IN_REALTIME:
-                    sleep(abs(tick_lengths_s - realtime_tick_length))
-                elif self.slowdown and realtime_tick_length < tick_lengths_s:
-                    # Simulation runs faster than real time but a slowdown was
-                    # requested
-                    tick_diff = tick_lengths_s - realtime_tick_length
-                    diff_slowdown = tick_diff * self.slowdown / SLOWDOWN_FACTOR
-                    log.trace("Slowdown: %.4f", diff_slowdown)
-                    if console is not None:
-                        self._handle_input(console, diff_slowdown)
+                    tick_runtime_s = time() - tick_time_counter
+                    sleep(abs(tick_length_s - tick_runtime_s))
+                elif self.slot_length_realtime:
+                    current_expected_time = \
+                        current_expected_time.add(seconds=tick_length_realtime_s)
+                    sleep_time = current_expected_time.timestamp() - now().timestamp()
+                    if sleep_time > 0:
+                        slow_down_applied = True
+                        sleep(sleep_time)
```

Comment:
I think it would be great to have a debug log here, that would denote the amount of sleep time that was forced to be used in order to sync with the desired time.

Reply:
ok! Agreed, I would omit the flag than (https://github.com/gridsingularity/d3a/pull/1011#discussion_r559716207)
OK?

Reply:
I'd very much prefer that, thanks!